test/deploy.py

```diff
@@ -0,0 +1,486 @@
+                'User {} is not in group {}. \nnewgrp {} \nsudo usermod -aG {} {}'
+                .format(self.usr, group_name, group_name, self.usr, group_name)
+            )
+
+    def checkLXDInitialised(self) -> None:
+        '''Check if LXD default profile contains network and root device.'''
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        default_devices = ['eth0', 'root']
+        devices = self.client.profiles.get('default').devices
+        if not all(device in devices for device in default_devices):
+            #if both default initialised network and root device do not exist.
+            raise AssertionError("LXD not initialised, please use 'lxd init --auto'")
+
+    def _get_random_string(self, length: int) -> string:
+        ''' Get a randomised string of given lentgh.'''
+        return ''.join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+    def createStoragePool(self, driver='dir', pool_name=test_tag) -> None:
+        ''' Create storage pool for LXD. '''
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        if not self.client.storage_pools.exists(pool_name):
+            config = {
+                "name": pool_name,
+                "driver": driver
+            }
+            print("Creating Storage Pool {}".format(pool_name))
+            self.client.storage_pools.create(config)
+            self.model['storage_pool'] = pool_name
+
+    def createVmProfile(
+        self, volumes: list, pool_name=test_tag, profile_name=test_tag,
+        repo_src=cwd
+    ) -> None:
+        ''' Create a VM profile for LXD'''
+        if not self.client:
+            raise AssertionError("LXD Client not available to runner.")
+
+        # If profile exists, it is expected to be already configured.
+        if not self.client.profiles.exists(profile_name):
+            # Load Profile yaml
+            with open(self.cwd+"/test/profile.yaml", 'r') as profile:
+                config = yaml.safe_load(profile.read())
+
+            devices = config['devices']
+            # # Patch pool name for root device
+            # devices['root']['pool'] = pool_name
+
+            # Patch block devices in the profile
+            for volume in volumes:
+                devices[volume] = {
+                    "pool": pool_name,
+                    "source": volume,
+                    "type": "disk",
+                }
+            #TODO: Add a print for the profile content for debugging.
+            # Create profile.
+            print("Creating VM Profile {}".format(profile_name))
+            self.client.profiles.create(
+                profile_name, config["config"], config["devices"]
+            )
+            self.model['profile'] = profile_name
+
+    def createVm(
+        self, image='ubuntu/jammy', flavor='c4-m10',
+        pool_name=test_tag, profile_name=test_tag, isStart=True
```

Comment:
Style nit: suggest using `is_start` for consistency
Similarly in below method args

Reply:
ack